### PR TITLE
Eccs add delete

### PR DESCRIPTION
**Pull request by claudiu-cc**
### Description
- Added the Delete endpoint to the ECCS client

### Parent Issue
Issue #190

### Related Pull Requests
PR #207

### Type(s) of Change (Split the PR if you check many)
- [x] Added user feature
- [ ] Added technical feature
- [x] Added tests
- [ ] Refactor
- [ ] Bugfix
- [x] Updated documentation
- [ ] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [x] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [ ] I have updated relevant workflows and deployment tools.
- [x] I have updated/added relevant documentation (readme, doc comments, etc).
- [x] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: applications/ECCS/app/delete.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"eccs/utils"
+	"log"
```
Comment (philip-cc) [inline applications/ECCS/app/delete.go:19]: Same comment as the other PR.

File: applications/ECCS/cmd/root.go [hunk 2]
```diff
@@ -83,6 +84,18 @@ var retrieveCmd = &cobra.Command{
...
 
+var deleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Deletes a stored object",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := app.Delete(userAT, objectID)
+		if err != nil {
+			return err
+		}
+		return nil
```
Comment (philip-cc) [inline applications/ECCS/cmd/root.go:95]:
```suggestion
		return app.Delete(userAT, objectID)
```

File: applications/ECCS/cmd/root.go [hunk 3]
```diff
@@ -123,7 +136,7 @@ var createUserCmd = &cobra.Command{
 	Use:   "createuser",
 	Short: "Creates a user on the server",
 	RunE: func(cmd *cobra.Command, args []string) error {
-		err := app.CreateUser(userAT, scopeRead, scopeCreate, scopeIndex, scopeObjectPermissions, scopeUserManagement)
+		err := app.CreateUser(userAT, scopeRead, scopeCreate, scopeDelete, scopeIndex, scopeObjectPermissions, scopeUserManagement)
```
Comment (philip-cc) [inline applications/ECCS/cmd/root.go:139]: Same comment as the other PR.

Comment (claudiu-cc) [reply applications/ECCS/cmd/root.go:135]: Fixed in the other PR #207

File: applications/ECCS/app/delete.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	if err != nil {
+		log.Fatalf("%v: %v", utils.Fail("Delete failed at NewClient"), err)
+	}
+
+	// Call Encryptonize and delete the object
+	err = client.Delete(objectID)
+	if err != nil {
+		log.Fatalf("%v: %v", utils.Fail("Delete failed at Delete"), err)
+	}
+
```
Comment (philip-cc) [inline applications/ECCS/app/delete.go:36]: Just noticed that we should have a log message along the lines of "succesfully deleted...", like we have for the other methods. The same goes for the update method.

**Review by philip-cc — APPROVED** [review]
Ship it!

**Pull request merged**